core/src/main/java/org/elasticsearch/search/suggest/term/TermSuggestionBuilder.java

```diff
@@ -65,12 +72,20 @@ public TermSuggestionBuilder(String name) {
      * tokens in the suggest text.
      * </ol>
      */
-    public TermSuggestionBuilder suggestMode(String suggestMode) {
+    public TermSuggestionBuilder suggestMode(@Nullable SuggestMode suggestMode) {
```

Comment:
This is a very good point about not allowing the builder to be in an invalid state.